test/unit/notifications.test.ts

```diff
@@ -109,6 +109,111 @@ describe('Notifications', () => {
+        expect(axiosMock.history.put[0].url).not.toContain('?ignoreErrors');
+      });
+      it('should fail all subscribe() calls from the same debounce interval with the same 403 error if one fails', async () => {
+        let aResult: undefined | 'fulfilled' | 'rejected';
+        let bResult: undefined | 'fulfilled' | 'rejected';
+        let cResult: undefined | 'fulfilled' | 'rejected';
+        await Promise.all([
+          notification.subscribe('a.topic').then(() => { aResult = 'fulfilled'; }, () => { aResult = 'rejected'; }),
+          notification.subscribe('b.topic').then(() => { bResult = 'fulfilled'; }, () => { bResult = 'rejected'; }),
+          notification.subscribe('c.topic').then(() => { cResult = 'fulfilled'; }, () => { cResult = 'rejected'; }),
+        ]);
+        // All subscribe() calls reject because the bulk subscribe response is 403 due to b.topic
+        expect(aResult).toEqual('rejected');
+        expect(bResult).toEqual('rejected');
+        expect(cResult).toEqual('rejected');
+        // But all 3 topics should remain in bulkSubscriptions
+        expect(Object.keys(notification.bulkSubscriptions)).toContain('a.topic');
+        expect(Object.keys(notification.bulkSubscriptions)).toContain('b.topic');
+        expect(Object.keys(notification.bulkSubscriptions)).toContain('c.topic');
+      });
+    });
+
+    describe('when enabled', () => {
+      beforeEach(() => {
+        notification = new Notifications(client, { host: 'localhost', enablePartialBulkResubscribe: true });
+        notification.stanzaInstance = getFakeStanzaClient();
+      });
+      it('should add ?ignoreErrors=true to bulk resubscribe', async () => {
+        await notification.subscribe('a.topic', () => {});
+        expect(axiosMock.history.put.length).toEqual(1);
+        expect(axiosMock.history.put[0].url).toContain('?ignoreErrors');
+      });
+      it('should individually succeed or fail subscribe() calls from the same debounce interval based on API response', async () => {
+        let aResult: undefined | 'fulfilled' | 'rejected';
+        let bResult: undefined | 'fulfilled' | 'rejected';
+        let cResult: undefined | 'fulfilled' | 'rejected';
+        await Promise.all([
+          notification.subscribe('a.topic').then(() => { aResult = 'fulfilled'; }, () => { aResult = 'rejected'; }),
+          notification.subscribe('b.topic').then(() => { bResult = 'fulfilled'; }, () => { bResult = 'rejected'; }),
+          notification.subscribe('c.topic').then(() => { cResult = 'fulfilled'; }, () => { cResult = 'rejected'; }),
+        ]);
+        expect(aResult).toEqual('fulfilled');
+        // b.topic subscribe() call is rejected because bulk subscribe result is "Rejected" for that topic
+        expect(bResult).toEqual('rejected');
+        // c.topic subscribe() call is rejected because bulk subscribe result is "Unknown" for that topic (not in the API response)
+        expect(cResult).toEqual('rejected');
```

Comment:
It looks like with `ignoreErrors=true` and a topic that will be rejected (e.g. `b.topic`), the subscription attempt runs through the topic that will be rejected (e.g. attempt to subscribe to `a.topic`, attempt to subscribe to `b.topic`), then stops and the topics **after** the rejected topic aren't tried (e.g. the attempt to subscribe `c.topic` doesn't happen). Does that match what you've seen from the backend?

Reply:
Nah, the backend behavior is what you'd expect: all topics get processed regardless of order or other failures.

The stuff for `c.topic` here is really just to achieve 100% branch coverage. There's logic around if the API response doesn't include a topic we were expecting a result for and that's the case being tested by `c.topic`. I could spell that out in more code comments on the mocks above if you think it needs clarification.

Reply:
Ah! I was thrown off, so I wouldn't mind a little clarification in the code, but I'm also pretty satisfied with this being our record of the behavior. Or maybe add one more topic in the mock so we can show both aspects (all topics as processed **and** how we handle a topic not being included in the results)?

Reply:
See commit 445d4fd6 for clarified test behavior with more comments and test topics with more descriptive names... `a.ok.topic`, `b.bad.topic`, `c.unknown.topic`